cpp/src/io/json/nested_json_gpu.cu

```diff
@@ -1725,18 +1725,20 @@ std::pair<std::unique_ptr<column>, std::vector<column_name_info>> json_column_to
   return {};
 }
 
-table_with_metadata host_parse_nested_json(host_span<SymbolT const> input,
+table_with_metadata host_parse_nested_json(device_span<SymbolT const> d_input,
                                            cudf::io::json_reader_options const& options,
                                            rmm::cuda_stream_view stream,
                                            rmm::mr::device_memory_resource* mr)
 {
   // Range of orchestrating/encapsulating function
   CUDF_FUNC_RANGE();
 
-  auto const new_line_delimited_json = options.is_enabled_lines();
+  std::vector<SymbolT> h_input;
+  h_input.resize(d_input.size());
+  CUDF_CUDA_TRY(cudaMemcpyAsync(
+    h_input.data(), d_input.data(), h_input.size(), cudaMemcpyDefault, stream.value()));
```

Comment:
Should `h_input.size()` be `d_input.size() * sizeof(SymbolT)`? It looks like the factor of `sizeof(SymbolT)` is missing here.

If so, can this whole block be replaced with [`make_std_vector_async`](https://github.com/rapidsai/cudf/blob/8d1e2ccaccce86fcc9c9e71012137960c205bcf9/cpp/include/cudf/detail/utilities/vector_factories.hpp#L307)?

If that's _not_ the case and the size is correct, can we have a comment about why this is the correct size?
```suggestion
  auto h_input = cudf::detail::make_std_vector_async(d_input, stream);
```